internal/subscriber/retry/retrier_test.go

```diff
@@ -87,3 +105,68 @@ func Test_Interval_Calculator(t *testing.T) {
 		assert.Equal(t, test.want, actualIntervals)
 	}
 }
+
+func TestRetrier_Start(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	mockBrokerStore := mocks.NewMockIBrokerStore(ctrl)
+	mockCache := mocks2.NewMockICache(ctrl)
+	mockCache.EXPECT().Get(gomock.AssignableToTypeOf(ctx), gomock.Any()).Return([]byte{'0'}, nil).AnyTimes()
+	dcs := getMockDelayConsumer(ctx, ctrl)
+	producer := getMockProducer(ctx, ctrl)
+	mockBrokerStore.EXPECT().GetConsumer(
+		ctx,
+		gomock.Any(),
+	).Return(dcs, nil).AnyTimes()
+	mockBrokerStore.EXPECT().RemoveConsumer(ctx, gomock.Any()).AnyTimes()
+	mockBrokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+
+	r := &Retrier{
+		subscriberID:   "subscription-id",
+		subs:           getMockSubscriptionModel(),
+		bs:             mockBrokerStore,
+		ch:             mockCache,
+		backoff:        NewExponentialWindowBackoff(),
+		finder:         NewClosestIntervalWithCeil(),
+		handler:        NewPushToPrimaryRetryTopicHandler(mockBrokerStore),
+		delayConsumers: sync.Map{},
+		errChan:        make(chan error),
+	}
+	expDelayTopics := r.subs.GetDelayTopics()
+	sort.Strings(expDelayTopics)
+	err := r.Start(ctx)
+	assert.Nil(t, err)
+	dcTopics := []string{}
+	r.delayConsumers.Range(func(key, value interface{}) bool {
+		dc := value.(*DelayConsumer)
+		dcTopics = append(dcTopics, dc.topic)
+		return true
+	})
+	sort.Strings(dcTopics)
```

Comment:
@sumitkrzp What is this function sorting on based of?

Reply:
@akashgoyal96 Yes